WIP: manage a microservice application using traefik load balancer

allamand opened the pull request:
Hi,

I would like to add this Post showing how we can leverge the use of Traefik loadbalancer with docker swarm cluster.

With the recent change of scaling play-with-docker I didn't figure out how I can pre-set urls in the tutorial ideally the user could click on urls that can be changed dynamically like it is done on Katacoda.

If the content seams ok for you I can create more PR with : 
- Manage a microservice using vfarcic docker flow proxy
- Create a cassandra cluster on swarm
- Create a cassandra cluster on swarm and monitore it with Prometheus

Don't hesitate to comment on these.

Sébastien

allamand committed:
manage a microservice application using traefik load balancer

marcosnils commented:
@allamand yes, yes yes!. Thx a lot for this post, been wanting something like this for a while. 

> With the recent change of scaling play-with-docker I didn't figure out how I can pre-set urls in the tutorial ideally the user could click on urls that can be changed dynamically like it is done on Katacoda.

I don't quite understand this, the training site already supports populating links to exposing services like this: ` [This is a link](#){:data-term=".term1"}{:data-port="8888"}`. Is this what you need?


PS: Regarding content streams, please keep them coming!.

allamand commented:
Hi @marcosnil,

I've missed this feature, seams to be exactly what I was looking for. I'll test that asap, thanks!

Sébastien

> Le 31 mars 2017 à 18:40, Marcos Nils <notifications@github.com> a écrit :
> 
> @allamand yes, yes yes!. Thx a lot for this post, been wanting something like this for a while.
> 
> With the recent change of scaling play-with-docker I didn't figure out how I can pre-set urls in the tutorial ideally the user could click on urls that can be changed dynamically like it is done on Katacoda.
> 
> I don't quite understand this, the training site already supports populating links to exposing services like this: [This is a link](#){:data-term=".term1"}{:data-port="8888"}. Is this what you need?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

allamand commented:
@marcosnils that was definitely what I was looking for great job ;) I think I would like to extend it a little (addying uri in the end of the url) is that already possible ?

I was including this modification based on latest version on my pwd instance, but I get some errors in sdk.js  `var sessionData = JSON.parse(resp.responseText);`

My PWD return an instance id, while your PWD server return a json.. I try fetch scaling branch but still get an ID instead of a Json, how can I fix this ? (sorry to write thi in the PR log, I can move it if needed)

marcosnils commented:
> I think I would like to extend it a little (addying uri in the end of the url) is that already possible ?

I don't think so, but it makes totally sense. Maybe something like this could work `[This is a link](/my/uri){:data-term=".term1"}{:data-port="8888"}`.

> My PWD return an instance id, while your PWD server return a json.. I try fetch scaling branch but still get an ID instead of a Json, how can I fix this ? (sorry to write thi in the PR log, I can move it if needed)

Hmmm check your PWD instance again, we're returning a JSON from all the ajax `new_session` calls. https://github.com/franela/play-with-docker/blob/master/handlers/new_session.go#L37-L41

allamand commented:
@marcosnils the link does not works with URI.
Perhaps we could add a rewrite of the href of the link to the new target

marcosnils commented:
> the link does not works with URI.

@allamand when I say that **it could work**, I mean that we could implement it that way. It's not currently supported.

allamand commented:
Ok got it ;)

allamand committed:
allow to add URI in links to internal services and forbid link with (#) we must uses at least (/)

allamand commented:
@marcosnils I've added the possibility to make links with `[This is a link](/my/uri){:data-term=".term1"}{:data-port="8888"}`, and updated my 2 blog posts.

Note that link with [This is a link](#){:data-term=".term1"}{:data-port="8888"} won't works because the # is replaced with current location. We must used instead

[This is a link](/){:data-term=".term1"}{:data-port="8888"}

I've corrected earlier blog in the repo that was using that

marcosnils commented:
@allamand amazing!. I'll test it locally and merge!

marcosnils commented on _posts/2017-04-01-docker-flow-proxy.markdown:
```diff
@@ -0,0 +1,505 @@
+title:  "WebApps with Docker Flow Proxy"
+date:   2017-04-01 10:51:47 +0530
+author: "Sébastien Allamand (allamand)"
+categories: webapps
```
`categories` should be `category` and needs to be one of ["beginner", "intermediate", "advanced"]

marcosnils commented on _posts/2017-03-31-traefik-load-balancing.markdown:
```diff
@@ -0,0 +1,437 @@
+title:  "WebApps with Traefik LoadBalancing"
+date:   2017-03-31 10:51:47 +0530
+author: "Sébastien Allamand (allamand)"
+categories: webapps
```
`categories` should be `category` and needs to be one of ["beginner", "intermediate", "advanced"]

marcosnils reviewed (commented):
@allamand PR is amazing. Just added two minor tagging comments.

marcosnils commented:
@allamand I'll change this myself and merge to master as it's only a small detail.

allamand commented:
@marcosnils Thanks a lot It was my first merged PR ;)

marcosnils commented:
@allamand and it was awesome!. Please, keep them coming!.
